services/common/src/redux/slices/permitSearchSlice.ts

```diff
@@ -0,0 +1,102 @@
+            state.query = action.payload;
+        }),
+        setFilters: create.reducer((state, action: { payload: Array<{ category: string; value: string }> }) => {
+            state.filters = action.payload;
+        }),
+        searchPermitConditions: create.asyncThunk(
+            async (payload: SearchQuery, thunkApi) => {
+                thunkApi.dispatch(showLoading());
+                const headers = createRequestHeader();
+
+                try {
+                    const response = await CustomAxios().post(
+                        `${ENVIRONMENT.apiUrl}/search/permit-conditions`,
+                        payload,
+                        headers
+                    );
+
+                    thunkApi.dispatch(hideLoading());
+                    return response.data;
+                } catch (error) {
+                    thunkApi.dispatch(hideLoading());
+                    throw error;
+                }
+            },
+            {
+                pending: (state) => {
+                    state.loading = true;
+                },
+                fulfilled: (state, action) => {
+                    state.loading = false;
+                    state.results = action.payload;
+
+                    // Merge new facets with existing ones
+                    const currentFacets = action.payload.facets || {};
+
+                    state.allFacets = {
+                        ...state.allFacets,
+                        ...Object.fromEntries(
+                            Object.entries(currentFacets).map(([key, values]) => [
+                                key,
+                                Array.from(
+                                    new Map([
+                                        ...(state.allFacets[key] || []),
+                                        ...(values as Facet[])
+                                    ].map(item => [item.value, item])).values()
+                                )
+                            ])
+                        )
+                    };
+
+                    // Also update the results allFacets
+                    state.results.allFacets = state.allFacets;
+                },
+                rejected: (state) => {
+                    state.loading = false;
+                },
+            }
+        ),
+    }),
+});
+
+export const { searchPermitConditions, setQuery, setFilters } = permitSearchSlice.actions;
+
+// Type the selectors
+export const selectSearchQuery = (state: RootState): string => state.permitSearch.query;
+export const selectSearchFilters = (state: RootState): Array<{ category: string; value: string }> => state.permitSearch.filters;
+export const selectSearchResults = (state: RootState): SearchResult | null => state.permitSearch.results;
+export const selectSearchLoading = (state: RootState): boolean => state.permitSearch.loading;
+export const selectAllFacets = (state: RootState): { [key: string]: Facet[] } => state.permitSearch.allFacets;
```

Comment:
I'm wondering if there's any advantage obtained here by creating the selectors outside of the slice and passing the root state vs creating them in the selectors property of the slice and only passing them the slice state. My gut says that there might be more re-calculation by passing the root state but I'm not 100% sure of that.

Reply:
Nope, no real advantage - just forgot that the selector property on the slice was an option. Updated to move it inside the slice